### ISSUE TITLE
check_memory with check_nrpe buggy in 0.4.3.143 ?

### ISSUE THREAD

**Issue by MaskierterMuchacho** [mickem/nscp#307]
Hi, I'm using check_memory with check_nrpe and it seems not to work as expected.

This is the command how I use it
`check_nrpe - H $HOSTADDRESS -c check_memory -a --show-all "warning=used > 90%" "critical=used > 95%" "detail-syntax=RAM status: free=%(free), used=%(used), total size=%(size), type=%(type)"`

and this is what I get
`WARNING: comitted = 25.216GB, physical = 26.816GB|'committed'=25.21591GB;88.69779;99.78501;0;110.87223 'committed %='22%;79;89;0;100 'physical'=26.81603GB;25.5996;28.7995;0;31.9995 'physical %'=83%;79;89;0;100`

So few things seem to be wrong here:
1. warning is set to 90% and critical to 95%, but in the output it says 79 for warning and 89 for critical, but that's completlty wrong. why?
2. it returns a WARNING at 83% though the treshold is 90%

One more thing: If i remove the **--show-all** from the command, this is what I get:
Used command:
`check_nrpe - H $HOSTADDRESS -c check_memory -a "warning=used > 90%" "critical=used > 95%" "detail-syntax=RAM status: free=%(free), used=%(used), total size=%(size), type=%(type)"`

Outoput:
`OK: RAM status: free=85.776GB, used=25.096GB, total size=110.872GB, type=committed, RAM status: free=5.261GB, used=26.738GB, total size=32GB, type=physical|'committed'=25.0963GB;99.78501;105.32862;0;110.87223 'committed %'=22%;89;94;0;100 'physical'=26.73849GB;28.79955;30.39952;0;31.9995 'physical %'=83%;89;94;0;100`

1. warning is set to 90% and critical to 95% yet the treshold in the output are wrong (1 below defined tresholds) -> 89 and 94
2. why does the command has so massive problems when using --show-all ?

**Comment by box293** [comment]
Try show-all instead of **--**show-all

--show-all
```
check_nrpe -H win2008r2-01 -c check_memory -a --show-all "warn=used>90%" "crit=used>95%" "detail-syntax=RAM status: free=%(free), used=%(used), total size=%(size), type=%(type)"

committed = 811.164MB, physical = 861.508MB|'committed'=0.79215GB;6.39775;7.19747;0;7.99719 'committed %'=10%;80;90;0;100 'physical'=0.84131GB;3.1996;3.59955;0;3.9995 'physical %'=21%;80;90;0;100
```

show-all
```
check_nrpe -H win2008r2-01 -c check_memory -a show-all "warn=used>90%" "crit=used>95%" "detail-syntax=RAM status: free=%(free), used=%(used), total size=%(size), type=%(type)"

RAM status: free=7.205GB, used=811.195MB, total size=7.997GB, type=committed, RAM status: free=3.158GB, used=861.492MB, total size=4GB, type=physical|'committed'=0.79218GB;7.19747;7.59733;0;7.99719 'committed %'=10%;90;95;0;100 'physical'=0.8413GB;3.59955;3.79952;0;3.9995 'physical %'=21%;90;95;0;100

```

**Comment by MaskierterMuchacho** [comment]
Hi bix293,

thanks for your reply!
Unfortunally this solution still doesn't work.
Treshold are still wrong and also the output is completly the same with and without **show-all**

**Comment by mickem** [comment]
1, this is a know rounding issue which is solved in 0.5.0 (the second 1 is the same issue right?)

2, Using -- and non -- options is a bad idea.
The parser will look at the first option and then decided if your trying to use -- or not.
In this case my guess is that it finds -- and ignore the rest of the options as they are missing -- and falls back to the default warning which is 80%.

So the problem is I think mixing -- and non -- in front of options.

**Comment by MaskierterMuchacho** [comment]
Hi mickem, thanks for your reply!

Yes indeed. It seems to be a problem with the --
when I remove the -- the output will use the treshold (-1) that's given in the command correctly and doesn't use the default treshold. Good to know that it's a rounding issue.

Thank you!

**Comment by mickem** [comment]
Rounding issue is (BTW) resolved ion 0.5.0
